src/main/scala/firrtl/transforms/Dedup.scala

```diff
@@ -423,7 +427,11 @@ object DedupModules {
     val dedupedName2module = tag2name.map({ case (tag, name) => name -> DedupModules.dedupInstances(top, name, moduleMap, name2name, renameMap) })
 
     // Build map from original name to corresponding deduped module
-    val name2module = tag2all.flatMap({ case (tag, names) => names.map(n => n -> dedupedName2module(tag2name(tag))) })
+    // It is important to flatMap before looking up the DefModules so that they aren't hashed
```

Comment:
Makes sense, good idea waiting for the smaller `Map` to be realized by the concatenation.

Reply:
I see now that your first commit message implies that it does.

Reply:
I think your second comment is in the wrong place lol